forms/TreeDropdownField.php

```diff
@@ -460,15 +460,26 @@ protected function populateIDs() {
 				if ($row->ParentID) $parents[$row->ParentID] = true;
 				$this->searchIds[$row->ID] = true;
 			}
+			
+			$sourceObject = $this->sourceObject;
+			$qSuffix = '';
+			// The parent of a file can also be a file (and not necessarily a folder, see File::has_one()) but we are 
+			// only interested in displaying folders in the dropdown field, when searching for folders.
+			if ($sourceObject == 'Folder') {
+				$qSuffix .= '"ClassName" IN (\''.implode('\',\'',Convert::raw2sql(ClassInfo::subclassesFor('Folder'))).'\')';
```

Comment:
What is the need for this argument? Why is `Folder` special?

The query builder will automatically do this for `Folder::get()` just like it would for `Page::get()`.